Docs/source/Software/Tasks.rst

```diff
@@ -0,0 +1,41 @@
+********************************
+Real Time Operating System Tasks
+********************************
+
+Fault State Task: Manthan Upadhyaya
+===================================
+
+The Fault State Task is called when a fault condition is set off in the BPS. 
+These fault conditions can be found here(place link to fault conditions).
+
+Process:
+    1) All other tasks are not allowed to run unless they have are utilized sempahore. The other tasks will run until they post that semaphore and then they are prevented from running again. This occurs internally through the RTOS.
```

Comment:
I'm basically trying to say that if a thread is using a semaphore, that thread will run until the semaphore is posted. This is the thing Sijin was worried about but is internally done by the micrium rtos. I felt it should be included so it can be known that there is no instance where the fault state task will be stuck in a deadlock.

Reply:
I can reword if necessary.

Reply:
I don't think this is exactly how it works, since the scheduler is called periodically (in addition to whenever a semaphore is posted). For example, the scheduler may switch from the amperes thread to the watchdog thread whenever the correct amount of time has passed (regardless of if the amperes thread is posting a semaphore).

Reply:
I added the part with the scheduler, if it is unnecessary to include this I can just delete it also.